MediaAdminBundle/Tests/Functional/EventSubscriber/UpdateMediaReferenceSubscriberTest.php

```diff
@@ -0,0 +1,162 @@
+use OpenOrchestra\ModelBundle\Document\Status;
+use OpenOrchestra\ModelInterface\Event\StatusableEvent;
+use OpenOrchestra\ModelInterface\StatusEvents;
+use Symfony\Component\EventDispatcher\EventDispatcher;
+
+/**
+ * Class UpdateMediaReferenceSubscriberTest
+ */
+class UpdateMediaReferenceSubscriberTest extends AbstractControllerTest
+{
+    const ATTRIBUTE_ID_SUFFIX = "Id";
+    const METHOD_SUFFIX = "BlockConfiguration";
+
+    /**
+     * @var Node node
+     */
+    protected $node;
+
+    /**
+     * @var array medias
+     */
+    protected $medias;
+
+    /**
+     * @var Status pending
+     */
+    protected $pending;
+
+    /**
+     * @var Status published
+     */
+    protected $published;
+
+    /**
+     * @var EventDispatcher eventDispatcher
+     */
+    protected $eventDispatcher;
+
+    /**
+     * Set up the test
+     */
+    public function setUp()
+    {
+        parent::setUp();
+
+        $nodeRepository = static::$kernel->getContainer()->get('open_orchestra_model.repository.node');
+        $this->node = $nodeRepository->findOneByNodeIdAndLanguageAndSiteIdInLastVersion('root', 'en', '2');
+        $mediaRepository = static::$kernel->getContainer()->get('open_orchestra_media.repository.media');
+        $this->medias = array(
+            $mediaRepository->findOneByName("Image 03"),
+            $mediaRepository->findOneByName("Image 04")
+        );
+        $this->eventDispatcher = static::$kernel->getContainer()->get('event_dispatcher');
+        $statusRepository = static::$kernel->getContainer()->get('open_orchestra_model.repository.status');
+        $this->pending = $statusRepository->findOneByName('pending');
+        $this->published = $statusRepository->findOneByName('published');
+    }
+
+    /**
+     * @param array  $blockType
+     * @param int    $mediaIndex
+     *
+     * @dataProvider provideMediaBlocks
+     */
+    public function testAddMediaBlocks(array $blockType, $mediaIndex)
+    {
+        $this->node->setStatus($this->pending);
+        /** @var Media $media */
+        $media = $this->medias[$mediaIndex];
+        $this->checkMediaReference($media, null);
+
+        $block = $this->generateBlock($blockType, 'ET9reyt');
+        $this->node->addBlock($block);
+        $attributes = $block->getAttributes();
+        $attributes['pictures'] = array(MediaInterface::MEDIA_PREFIX . $media->getId());
+        $attributes['id'] = $blockType["component"] . ATTRIBUTE_ID_SUFFIX;
+        $method = $blockType["component"] . METHOD_SUFFIX;
+        $attributes = $this->$method($attributes);
+        $block->setAttributes($attributes);
+
+        $event = new StatusableEvent($this->node, $this->published);
+        $this->eventDispatcher->dispatch(StatusEvents::STATUS_CHANGE, $event);
+
+        $mediaBlockIndex = $this->node->getBlockIndex($block);
+        $expectedReference = "node-" . $this->node->getId() . "-" .  $mediaBlockIndex;
```

Comment:
this 'node' should also go into a class constant to be coherent with the other strings